File: bin/read_data.py
#!/usr/bin/python
import serial, sys, struct, time, datetime

device_filename = sys.argv[1]
ser = serial.Serial(device_filename, 9600, timeout=1)

# Find alignment
packet_size = 49
s = ser.read(100)
ser.read(packet_size - ((100 - s.find("start")) % packet_size))

last_millis = 0
handle = open("data/data_collect.dat", "w")

count = 0
start_time = time.time()
while start_time+1 > time.time():
	s = ser.read(49)
	data = struct.unpack("iffffffffff", s[5:])

	if(s[:5] != "start"):
		print("uh oh!", count)

	handle.write(str(datetime.datetime.now().microsecond))
	handle.write(" ")
	for j in range(0, 11):
		handle.write(str(data[j]))
		handle.write(" ")
	handle.write("\n")

	count = count + 1
	last_millis = data[0]

handle.close()
ser.close()
print("Done.")
print(count)

File: src/read_data.cpp
#include <stdlib.h>
#include <string.h>
#include <stdio.h>

int main(int argc, char** argv)
{
	FILE* handle = fopen(argv[1], "r");

	int packet_size = strlen("start") + strlen("end\n") + 4*11;
	char buffer[1000];

	fgets(buffer, 1000, handle);
	for(int i=0; i<10; i++)
	{
		printf("%s", buffer);
		fgets(buffer, 1000, handle);
	}

	fclose(handle);
	return 0;
}
